.gitlab-ci.yml

```diff
@@ -1252,7 +1254,11 @@ windows_choco_offline_7_x64:
   needs: ["fail_on_non_triggered_tag", "windows_msi_x64-a7"]
   variables:
     ARCH: "x64"
+  dependencies:
+    - windows_msi_x64-a7
```

Comment:
Why do you need to specify `windows_msi_x64-a7` both in needs and dependencies? What is the difference?

Reply:
- `needs` affects the order of execution: a job will wait for all jobs it needs before it starts. [1]
- `dependencies` affects which artifacts a jobs receives from previous stages. [2]

[1] https://docs.gitlab.com/ee/ci/yaml/#needs
[2] https://docs.gitlab.com/ee/ci/yaml/#dependencies

Reply:
Agreed with Pablo, they shouldn't be used together (it's deprecated since [Gitlab 12.6.0](https://docs.gitlab.com/ee/ci/yaml/#artifact-downloads-with-needs))

Reply:
I see, so it just limits the artifacts the job gets so that it is faster. Great, thank you!

Reply:
So `depends` implies that you only get those artifacts, then? TIL.

Reply:
Removed the `dependencies` fields and added an empty `artifacts` for the offline job which doesn't need the MSI.